browser/fragments/ably-epilogue.js

```diff
@@ -1,16 +1,35 @@
-if(typeof Realtime !== 'undefined') {
-	Ably.msgpack = msgpack;
-	Ably.Rest = Rest;
-	Ably.Realtime = Realtime;
-	Realtime.ConnectionManager = ConnectionManager;
-	Realtime.BufferUtils = Rest.BufferUtils = BufferUtils;
-	if(typeof(Crypto) !== 'undefined') Realtime.Crypto = Rest.Crypto = Crypto;
-	Realtime.Defaults = Rest.Defaults = Defaults;
-	Realtime.Http = Rest.Http = Http;
-	Realtime.Utils = Rest.Utils = Utils;
-	Realtime.Http = Rest.Http = Http;
-	Realtime.Message = Rest.Message = Message;
-	Realtime.PresenceMessage = Rest.PresenceMessage = PresenceMessage;
-	Realtime.ProtocolMessage = Rest.ProtocolMessage = ProtocolMessage;
-}
+	if(typeof Realtime !== 'undefined') {
+		Ably.msgpack = msgpack;
+		Ably.Rest = Rest;
+		Ably.Realtime = Realtime;
+		Realtime.ConnectionManager = ConnectionManager;
+		Realtime.BufferUtils = Rest.BufferUtils = BufferUtils;
+		if(typeof(Crypto) !== 'undefined') Realtime.Crypto = Rest.Crypto = Crypto;
+		Realtime.Defaults = Rest.Defaults = Defaults;
+		Realtime.Http = Rest.Http = Http;
+		Realtime.Utils = Rest.Utils = Utils;
+		Realtime.Http = Rest.Http = Http;
+		Realtime.Message = Rest.Message = Message;
+		Realtime.PresenceMessage = Rest.PresenceMessage = PresenceMessage;
+		Realtime.ProtocolMessage = Rest.ProtocolMessage = ProtocolMessage;
+	}
 }).call({});
+
+/* CommonJS support */
+if (typeof exports === "object" && exports) {
+	for (var obj in window.Ably) {
+		if (window.Ably.hasOwnProperty(obj)) {
+			exports[obj] = window.Ably[obj];
+		}
+	}
+	/* SystemJS support for default exports to be added to the root of the module
+	   https://github.com/frankwallis/plugin-typescript/issues/185 */
+	exports.__esModule = true;
+}
+
+/* AMD support */
+if (typeof define === "function" && define.amd) {
+  define("ably", [], function() {
+    return Ably;
+  });
+}
```

Comment:
> But if we're doing that, we should probs remove the separate ably-commonjs.js

True, although because we can't break backwards compatibility, the primary version always sets a global `Ably`, whereas the CommonJS one doesn't.  So I get what you're saying, but if we make `ably-commonj.sj` just use this ver then we're going to start polluting people's global namespace.